### PR TITLE
15.0 fix coupon access rights pim

### PR DESCRIPTION
**Pull request by pimodoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pimodoo** [comment]
@caburj here are two commits to fix access rights on coupons, could you have a look?

File: addons/sale_coupon/models/coupon_program.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ class CouponProgram(models.Model):
     # The api.depends is handled in `def modified` of `sale_coupon/models/sale_order.py`
     def _compute_order_count(self):
         for program in self:
-            program.order_count = self.env['sale.order.line'].search_count([('product_id', '=', program.discount_line_product_id.id)])
+            program.order_count = self.env['sale.order.line'].sudo().search_count([('product_id', '=', program.discount_line_product_id.id)])
```
Comment (Feyensv) [inline addons/sale_coupon/models/coupon_program.py:15]: Or set the count to 0 if the user doesn't have read access to SOL ?

Comment (pimodoo) [reply addons/sale_coupon/models/coupon_program.py:15]: @Feyensv Having a count to 0 will lead to error when you are using the "total_order_count" field to know if you have used the program more than the "maximum_use_number".

**Review by caburj — APPROVED** [review]

**Comment by pimodoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**